tests/foreman/cli/test_discoveryrule.py

```diff
@@ -103,20 +99,22 @@ def _create_discoveryrule(org, loc, hostgroup, options=None):
 
     @pytest.mark.tier1
     @pytest.mark.parametrize('name', **parametrized(valid_data_list()))
-    def test_positive_create_with_name(self, name, discoveryrule_factory, request, target_sat):
+    def test_positive_create_with_name(self, name, discoveryrule_factory, target_sat):
         """Create Discovery Rule using different names
 
         :id: 066e66bc-c572-4ae9-b458-90daf83bab54
 
-        :expectedresults: Rule should be successfully created
+        :expectedresults: Rule should be successfully created.
 
         :CaseImportance: Critical
 
         :parametrized: yes
         """
         rule = discoveryrule_factory(options={'name': name, 'priority': gen_int32()})
-        request.addfinalizer(target_sat.api.DiscoveryRule(id=rule.id).delete)
         assert rule.name == name
+        target_sat.api.DiscoveryRule(id=rule.id).delete()
```

Comment:
Why using API to delete it in CLI module? :)

Reply:
Right.